# HG changeset patch
# User "Yann E. MORIN" <yann.morin.1998@free.fr>
# Date 1511772629 -3600
#      Mon Nov 27 09:50:29 2017 +0100
# Branch yem/fixes
# Node ID 0848fa96c6eb13cf37249d317eff12cbd2f59ff7
# Parent  3d43b280298c39a67d1d889e01e173f52c12da35
utils: fix build with kernel headers >= 4.14

In kernel 4.14, CA_SET_PID was removed, in commit 833ff5e7feda (media:
ca.h: get rid of CA_SET_PID).

Fix dst-util to not fail the build when this is missing.

Fiuxes build failures such as:
    http://autobuild.buildroot.org/results/708/708f11809b2cafc2a3375dc515803c87b376ed4d/build-end.log
    http://autobuild.buildroot.org/results/e39/e3939d44376e92e8a35fb179d9890510334d8304/build-end.log

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff --git a/util/dst-utils/dst_test.c b/util/dst-utils/dst_test.c
--- a/util/dst-utils/dst_test.c
+++ b/util/dst-utils/dst_test.c
@@ -111,6 +111,7 @@
 	return 0;
 }
 
+#if defined CA_SET_PID
 static int dst_set_pid(int cafd)
 {
 	if ((ioctl(cafd, CA_SET_PID)) < 0) {
@@ -120,6 +121,7 @@
 
 	return 0;
 }
+#endif
 
 static int dst_get_descr(int cafd)
 {
@@ -230,8 +232,12 @@
 				dst_reset(cafd);
 				break;
 			case 'p':
+#if defined CA_SET_PID
 				printf("%s: PID\n", __FUNCTION__);
 				dst_set_pid(cafd);
+#else
+				printf("%s: PID not supported\n", __FUNCTION__);
+#endif
 				break;
 			case 'g':
 				printf("%s: Get Desc\n", __FUNCTION__);
